src/modelutils.py

```diff
@@ -28,20 +28,25 @@ def skip(*args, **kwargs):
 
 def dispatch_quantized_model(model):
     num_devices = torch.cuda.device_count()
-    device_map = {"model.embed_tokens": 0, "model.norm": num_devices - 1, "lm_head": num_devices - 1}
+    device_map = {"model.embed_tokens": 0, "model.norm": num_devices - 1, "lm_head": 0}
```

Comment:
`head` is now on 0th device in order to simplify things a bit. If you have better idea - I'll be glad to do better